src/private/fnft__misc.c

```diff
@@ -320,3 +321,82 @@ UINT misc_nextpowerof2(const UINT number)
         result *= 2;
     return result;
 }
+
+INT misc_resample(const UINT D, const REAL eps_t, COMPLEX const * const q,
+    const REAL delta, COMPLEX *const q_new)
+{
+    if (q == NULL)
+        return E_INVALID_ARGUMENT(q);
+    if (D <= 2)
+        return E_INVALID_ARGUMENT(D);
+    if (q_new == NULL)
+        return E_INVALID_ARGUMENT(q_new);
+    if (eps_t == 0)
+        return E_INVALID_ARGUMENT(eps_t);
+
+    fft_wrapper_plan_t plan_fwd = fft_wrapper_safe_plan_init();
+    fft_wrapper_plan_t plan_inv = fft_wrapper_safe_plan_init();
+    COMPLEX *buf0 = NULL, *buf1 = NULL;
+    REAL *freq = NULL;
+    INT ret_code;
+    UINT i;
+    // Allocate memory
+
+
+    const UINT lenmem = D * sizeof(COMPLEX);
+    buf0 = fft_wrapper_malloc(lenmem);
+    buf1 = fft_wrapper_malloc(lenmem);
+    freq = malloc(D * sizeof(REAL));
+    if (buf0 == NULL || buf1 == NULL || freq == NULL) {
+        ret_code = E_NOMEM;
+        goto release_mem;
+    }
+
+    ret_code = fft_wrapper_create_plan(&plan_fwd, D, buf0, buf1, -1);
+    CHECK_RETCODE(ret_code, release_mem);
+
+
+    ret_code = fft_wrapper_create_plan(&plan_inv, D, buf0, buf1, 1);
+    CHECK_RETCODE(ret_code, release_mem);
+
+    // Continuing the signal periodically
+    for (i = 0; i < D; i++)
+        buf0[i] = q[i];
+
+    ret_code = fft_wrapper_execute_plan(plan_fwd, buf0, buf1);
+    CHECK_RETCODE(ret_code, release_mem);
+
+    // Check that the signal spectrum decays sufficiently to ensure accurate interpolation
+    REAL tmp = CABS(buf1[0]);
+    for (i = 1; i<D; i++)
+        if (CABS(buf1[i]) > tmp)
+           tmp = CABS(buf1[i]);
+    if (CABS(buf1[D/2-1])/tmp > SQRT(EPSILON) || CABS(buf1[D/2+1])/tmp > SQRT(EPSILON))
```

Comment:
This criterion seems to become less reliable as the number of samples increases.

Maybe we could improve it a bit. Let's assume we apply a perfect low-pass filter to q(t) at, say, 90% of the total bandwidth, leading to q_lp(t). 

I think we could compute the error ||q-q_lp||/||q||, which has a clear interpretation, directly in the frequency domain using Parseval's relation. We could warn if that is `>SQRT(EPSILON)`.

Reply:
Indeed this seems to be a better criterion. I will implement it now.